src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -240,21 +240,30 @@ async Task<DurableHttpResponse> IDurableOrchestrationContext.CallHttpAsync(Durab
 
             while (currStatusCode == HttpStatusCode.Accepted && req.AsynchronousPatternEnabled)
             {
-                Dictionary<string, StringValues> headersDictionary = new Dictionary<string, StringValues>(durableHttpResponse.Headers);
+                var headersDictionary = new Dictionary<string, StringValues>(
+                        durableHttpResponse.Headers,
+                        StringComparer.OrdinalIgnoreCase);
+
                 DateTime fireAt = default(DateTime);
                 if (headersDictionary.ContainsKey("Retry-After"))
                 {
-                    fireAt = this.InnerContext.CurrentUtcDateTime.AddSeconds(int.Parse(headersDictionary["Retry-After"]));
+                    fireAt = this.InnerContext
+                                .CurrentUtcDateTime
```

Comment:
nit: Could we keep .CurrentUtcDateTime on the first line? It's a little weird to have hanging property references like this.